test/python/test_deeplabv3p_ar_detect.py

```diff
@@ -0,0 +1,109 @@
+try:
+    from mpi4py import *
+    rank = MPI.COMM_WORLD.Get_rank()
+    n_ranks = MPI.COMM_WORLD.Get_size()
+except:
+    rank = 0
+    n_ranks = 1
+import os
+import sys
+import numpy as np
+from teca import *
+
+
+def get_padding_params(n_lat, n_lon):
```

Comment:
the padding parameter are specific to your neural network. these are essentially "magic" numbers that belong to the given network. this is a detail that should be private part of the implementation. we can see the trouble that putting them here is causing below - in line 53 where you have to run the pipeline once to get the lat, lon dimensions. I believe also that you are now struggling to deal with the output image being different dimensions than the input since output of the NN is not padded. There are of course ways to propagate that info through the pipeline but I think that because padding dimension are an internal detail specific to each NN and hence should be private that you should run the padding and layering algorithms inside the execute of `deeplabv3p_ar_detect` where the padding parameters are known. Better yet simply manipulate the arrays directly before passing control to the `teca_model_segmentation`.